numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -4435,6 +4436,13 @@ PyArray_DescrFromType(int type)
         return NULL;
     }
     else if ((type == NPY_CHAR) || (type == NPY_CHARLTR)) {
```

Comment:
The `CHAR_LTR` refers to the `dtype(...).char` attribute, which is the only difference from the string type.
```
In [1]: dtype('c') == dtype('S1')
Out[1]: True
```